packages/consensus/src/types/utreexo.cairo

```diff
@@ -111,6 +112,49 @@ pub impl UtreexoAccumulatorImpl of UtreexoAccumulator {
         self.num_leaves += 1_u64;
     }
 
+    /// Removes an output from the accumulator.
+    /// https://eprint.iacr.org/2019/611.pdf Algorithm 2 DeleteOne
+    fn delete(ref self: UtreexoState, proof: @UtreexoProof) {
+        let mut roots: Array<Option<felt252>> = array![];
```

Comment:
Instead, it's better to construct the resulting array of roots, similar to how addition is implemented.